blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -58,8 +61,15 @@
     private static final String dataTypeCacheKey = MetricMetadata.TYPE.toString().toLowerCase();
 
     private static final Keyspace keyspace = getKeyspace();
+    private static int enumThreadCount = Configuration.getInstance().getIntegerProperty(CoreConfig.ENUM_READ_THREADS);
+    private ExecutorService taskExecutor =  null;
 
     public static AstyanaxReader getInstance() {
+        if (INSTANCE.taskExecutor == null || INSTANCE.taskExecutor.isShutdown()) {
+            INSTANCE.taskExecutor = new ThreadPoolBuilder().withUnboundedQueue()
+                    .withCorePoolSize(enumThreadCount)
+                    .withMaxPoolSize(enumThreadCount).withName("Retrieving Enum Values").build();
+        }
         return INSTANCE;
```

Comment:
I like how you made the threadpool a class instance here.  We're definite we'll get only on instance of taskExecutor across all AtyanaxReader instances?